### PR TITLE
Improvement/mobile menu

### PR DESCRIPTION
**Pull request by lmeri**
- Added menu for mobile devices
-> Mobile menu is used when the device's screen size is smaller or equal to 760px. It is checked when first navigating to the app.
- Sorted job postings first by isHidden status and then alphabetically
- Fixed promblems related to Header
- In job posting's applicants view changed styles of the "back to job posting" button.
-> This fixes errors related to LinkButton requiring text prop.

### CODE REVIEW

File: client/src/components/Header.js [hunk 1]
```diff
@@ -1,32 +1,62 @@
 import React, { Component } from 'react'
 import { Link } from 'react-router-dom'
+import { withRouter } from 'react-router'
 import { connect } from 'react-redux'
 import PropTypes from 'prop-types'
 
+import { ExitToApp } from '@material-ui/icons'
 import { ActionButton, LinkButton } from './Buttons'
```
Comment (llmlks) [inline client/src/components/Header.js]: ActionButton is not used anywhere, import can be removed

File: client/src/components/Header.js [hunk 2]
```diff
@@ -1,32 +1,62 @@
 import React, { Component } from 'react'
 import { Link } from 'react-router-dom'
+import { withRouter } from 'react-router'
 import { connect } from 'react-redux'
 import PropTypes from 'prop-types'
 
+import { ExitToApp } from '@material-ui/icons'
```
Comment (llmlks) [inline client/src/components/Header.js]: Same as with `Fix/dispatching-application` [information about how to minimize bundle size](https://material-ui.com/guides/minimizing-bundle-size/)

File: client/src/components/MobileMenu.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import React, { Component } from 'react'
+import { NavLink } from 'react-router-dom'
+import { withRouter } from 'react-router'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+
+import { Menu, ExitToApp } from '@material-ui/icons'
```
Comment (llmlks) [inline client/src/components/MobileMenu.js]: [information about how to minimize bundle size](https://material-ui.com/guides/minimizing-bundle-size/)

File: client/src/components/MobileMenu.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    }
+  }
+
+  componentDidUpdate(pProps) {
+    if (pProps.location !== this.props.location) {
+      this.setState({ isEnlarged: false })
+    }
+  }
+
+  onMenuClick = () => {
```
Comment (llmlks) [inline client/src/components/MobileMenu.js]: Currently clicking outside the menu does not close the dropdown

**Comment by lmeri** [comment]
The menu works for me on several browsers, so like we spoke let's check how it looks like on staging.
Oh, and the activeStyle will work after I rebase this branch to master. 🙂

**Pull request merged**